docs/source/tracking.rst

```diff
@@ -76,6 +76,104 @@ There are different kinds of remote tracking URIs:
   `[Azure] <http://docs.microsoft.com/azure/databricks/applications/mlflow/access-hosted-tracking-server>`_, or :ref:`the quickstart <quickstart_logging_to_remote_server>` to
   easily get started with hosted MLflow on Databricks Community Edition.
 
+How Runs and Artifacts are Recorded
+===================================
+As mentioned above, MLflow runs can be recorded to local files, to a SQLAlchemy compatible database, or remotely to a tracking server.
+For storing runs and artifacts, MLflow uses two components for `storage <https://mlflow.org/docs/latest/tracking.html#storage>`_: backend store and artifact store. While the backend store persists
+MLflow entities (runs, parameters, metrics, tags, notes, metadata, etc), the artifact store persists artifacts
+(files, models, images, in-memory objects, or model summary, etc).
+
+Which `backend store <https://mlflow.org/docs/latest/tracking.html#backend-stores>`_ or `artifact store <https://mlflow.org/docs/latest/tracking.html#artifact-stores>`_ is used by the MLFlow client or tracking server
+depends on four different scenarios.
+
+Scenario 1: MLflow on the localhost
+-----------------------------------
+
+A common scenario among developers who want to develop locally, MLflow runs on a localhost or a laptop, where both the
+backend store and artifact store is a directory on the local filesystem ``./mlruns,`` as shown in the diagram.
+The `MLflowClient` directly interfaces with an instance of a `FileStore` and `LocalArtifactRepository`.
+
+.. figure:: _static/images/scenario_1.png
+
+In this simple scenario, the `MLFlowClient` is responsible to create instances of respective stores and
+interfaces with them to record MLflow entities and artifacts:
+
+ * `MLflowClient` creates an instance of a `LocalArtifactRepository` (to store artifacts)
+ * `MLflowClient` creates an instance of a `FileStore` (to save MLflow entities)
+
+Scenario 2: MLflow on the localhost with SQLite
+-----------------------------------------------
+
+A slight variation from scenario 1, MLflow runs on a localhost or laptop but with backend store as an SQLAlchemy compatible
+database type: **SQLite**. Artifacts are stored under the local ``./mlruns`` directory, and MLflow entities are
+inserted in an SQLite database file ``mlruns.db``.
+
+.. figure:: _static/images/scenario_2.png
+
+As above, `MLflowClient` is responsible to create respective instances:
```

Comment:
In this scenario, the MLflow client uses the following interfaces to record MLflow entities and artifacts:

Reply:
done!